src/base/LruMap.h

```diff
@@ -148,39 +169,47 @@ LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
     if (memLimit() == 0)
         return false;
 
-    index.push_front(new Entry(key, t));
+    auto e = new Entry(key, t, ttl);
+    memUsed_ += e->memCost() + e->key.length() + sizeof(Entry) + sizeof(MapPair);
+    index.push_front(e);
     storage.insert(MapPair(key, index.begin()));
 
     ++entries_;
+
+    // if over memory limit, start purging
+    while (memUsed_ > memLimit_)
```

Comment:
Given that this code is an LRU, the answers to the first two are obviously yes. If we ignore entries simply for their size it becomes a GDS algorithm, not a basic LRU.

Anyhow, re-design of the basic algorithm is out of scope for this PR. The changes to LruMap are being forced in order to retain critical security properties around credential storage.